<launch>
    
    <arg name="node_start_delay" default="30.0" />  

    <rosparam command="load" file="$(find swarm)/config/crazyflies.yaml"/>
    <rosparam command="load" file="$(find swarm)/config/algorithm_params.yaml"/>
    <rosparam command="load" file="$(find swarm)/config/obstacles.yaml"/>


    <node pkg="swarm" type="simulation_setup.py" name="simulation_setup" output="screen"/>


    <!--node name="obstacle_spawn" pkg="gazebo_ros" type="spawn_model" output="screen" args="-urdf -file /home/emirhan/Swarm/src/Swarm-lib/launch/obstacle.urdf -model my_obstacle  -x 0 -y 0 -z 0"/-->
 
    <node pkg="swarm" type="takeoff_iris_multiple.py" name="takeoff" output="screen"/>

    <node pkg="swarm" type="iris_controller.py" name="iris_controller" output="screen"/>

    <node pkg="swarm" type="mission_planner.py" name="mission_planner" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />

</launch>